src/aspects/ec2-automated-shutdown/index.ts

```diff
@@ -0,0 +1,142 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { join } from 'path';
+import { Stack, Duration, IAspect } from 'aws-cdk-lib';
+import { Metric, Alarm, ComparisonOperator } from 'aws-cdk-lib/aws-cloudwatch';
+import { SnsAction } from 'aws-cdk-lib/aws-cloudwatch-actions';
+import { Instance, CfnInstance } from 'aws-cdk-lib/aws-ec2';
+import { PolicyStatement, Effect, AnyPrincipal } from 'aws-cdk-lib/aws-iam';
+import { IKey } from 'aws-cdk-lib/aws-kms';
+import { Code, Runtime } from 'aws-cdk-lib/aws-lambda';
+import { SnsEventSource } from 'aws-cdk-lib/aws-lambda-event-sources';
+import { Topic } from 'aws-cdk-lib/aws-sns';
+import { IConstruct } from 'constructs';
+import { SecureFunction } from '../../constructs/secure-function';
+import { LambdaConfiguration } from '../../interfaces/lambda-configuration';
+
+export interface EC2ShutdownProps {
+    /**
+     * The % CPU threshold below which the instance will be shut down
+     */
+    readonly cpuThreshold: number;
```

Comment:
Instead of forcing the Metric to be CPU - can we make this an optional parameter called `metric` that accepts a Metric that the user creates? If they don't supply one, we can use CPU at a predefined setting that you think is best. That way the user can use other Metrics that they may be interested in for the thing that shuts it down